src/main/java/com/aws/greengrass/localdebugconsole/SimpleHttpServer.java

```diff
@@ -154,13 +218,53 @@ public void startup() throws InterruptedException {
         reportState(State.RUNNING);
     }
 
+    static String fingerprintCert(X509Certificate cert)
+            throws NoSuchAlgorithmException, CertificateEncodingException {
+        StringBuilder sb = new StringBuilder();
+        for (byte b : MessageDigest.getInstance("SHA-256").digest(cert.getEncoded())) {
+            sb.append(String.format("%02X ", b));
+        }
+        return sb.toString().trim();
+    }
+
+    private static X509Certificate selfSign(KeyPair keyPair, String subjectDN) throws OperatorCreationException,
+            CertificateException,
+            IOException
+    {
+        java.security.Provider bcProvider = new BouncyCastleProvider();
+        Security.addProvider(bcProvider);
+
+        long now = System.currentTimeMillis() - TimeUnit.SECONDS.toMillis(60);
+        Date startDate = new Date(now);
+
+        X500Name dnName = new X500Name("cn=" + subjectDN);
+        BigInteger certSerialNumber = new BigInteger(Long.toString(now)); // Using the current timestamp as the certificate serial number
+
+        Calendar calendar = Calendar.getInstance();
+        calendar.setTime(startDate);
+        calendar.add(Calendar.YEAR, 1); // 1 Year validity
+        Date endDate = calendar.getTime();
+
+        String signatureAlgorithm = "SHA256WithRSA";
+        ContentSigner contentSigner = new JcaContentSignerBuilder(signatureAlgorithm).build(keyPair.getPrivate());
+        JcaX509v3CertificateBuilder
+                certBuilder = new JcaX509v3CertificateBuilder(dnName, certSerialNumber, startDate,
+                endDate, dnName, keyPair.getPublic());
+        BasicConstraints basicConstraints = new BasicConstraints(false);
+        certBuilder.addExtension(new ASN1ObjectIdentifier("2.5.29.19"), true, basicConstraints);
```

Comment:
What is `2.5.29.19`?

Reply:
https://www.alvestrand.no/objectid/2.5.29.19.html

Basic constraints OID. It is required to be recognized by browsers as a proper certificate.